ambari-agent/src/main/python/ambari_agent/FileCache.py

```diff
@@ -247,7 +247,7 @@ def write_hash_sum(self, directory, new_hash):
     try:
       with open(hash_file, "w") as fh:
         fh.write(new_hash)
-      os.chmod(hash_file, 0o666)
+      os.chmod(hash_file, 0o755)
```

Comment:
Why 755 and not 644?  Are these files executable?  Should the be executable by all?  Maybe 640 or 600 is in order here?

Reply:
Hi @rlevas ,

Thanks for your time reviewing the code,
The files are not executable , 
but i see in ambari-server we are giving the permission 755 for has files , 
code ref : https://github.com/apache/ambari/blob/5b36cdfd87b756eba922dfd1ac5419552f4d375f/ambari-server/src/main/python/ambari_server/resourceFilesKeeper.py
 
and i thought not to change the permissions defined by ambari-server and keep the same.

Let me know your thoughts, IMHO its not executable files.

Reply:
Maybe both places need to be fixed.  

Do we know who/what needs to read these hash files?  Since this file would contain only a hash I guess it can be world-readable without any issues.  However I can imagine some 
security audit to flag this and am leaning towards closing up access to this file as much as possible. I say we start with 640, or 600 if we are paranoid.  Then open it up further if we really need to.

@aonishuk , @mpapirkovskyy , @jonathan-hurley... any thoughts on this?